src/proxy/reconnect.rs

```diff
@@ -149,6 +219,15 @@ where
                     debug!("connect error to {:?}: {}", self.target, err);
                 }
 
+                // Set a backoff if appropriate.
+                //
+                // This future need not be polled immediately because the
+                // task is notified below.
+                self.active_backoff = match self.backoff {
+                    Backoff::None => None,
+                    Backoff::Fixed(ref wait) => Some(Delay::new(clock::now() + *wait)),
```

Comment:
Since `Duration` is `Copy`, this doesn't need to be a `ref` and de-reference, just `BackoffFixed(wait)`.